docs/enhancements/1-crds-and-controllers.md

```diff
@@ -0,0 +1,430 @@
+| Status | Authors   | Created    | 
+|--------|-----------|------------|
+| Review | @jewertow | Jan 2 2025 |
+
+# CRDs and controllers
+
+## Overview
+
+Currently, we use helm to manage the controller deployment and its configuration. The configuration is passed to the controller as CLI arguments.
+This approach requires restarting the controller on any configuration change and causes pod failures in case of a misconfiguration.
+
+Additionally, current approach to resource management is not reliable. We apply resources when it's necessary using the Istio client's `Apply` function,
+so we are not able to reconcile created objects on user's modifications or controller uninstallation.
+
+### Goals
+
+Design CRDs for better user experience and ensure proper resource lifecycle and garbage collection.
```

Comment:
I was thinking we can expand the goals based on what has been mentioned in the subsequent sections as well as overview, which is focused on drawbacks of existing approach.

```suggestion
We aim to improve the reliability and flexibility of Service Mesh Federation by:

- Introducing CRDs for a better user experience and declarative resource management.
- Ensuring proper resource lifecycle management.
- Preventing configuration drift by detecting and correcting unintended modifications.
```